### PR TITLE
Source category selection popup (Fix #765)

### LINKED ISSUE #765: Group source lists by category
The source list is becoming insanely long now, would be good to group things like in the Adjust GVx custom function, for example:
Inputs
Controls
LS
Channels
Trainer inputs
GVARs
Telemetry

Hmm now that's actually a whole lot of categories to browse through too :(

### PR DESCRIPTION
**Pull request by kilrah**

### CODE REVIEW

**Comment by kilrah** [comment]
@mhotar, up for some graphics? ;)

screenshot-1

**Comment by bsongis** [comment]
Cool! Strange there is no scroll line on the 1st screenshot?

**Comment by projectkk2glider** [comment]
This was badly needed! The list was getting very loooong... :+1:

File: radio/src/gui/Taranis/helpers.cpp [hunk 1]
```diff
@@ -347,3 +347,14 @@ bool modelHasNotes()
...
 }
 
+int getFirstAvailableSource(int min, int max, bool (*func)(int))
+{
+  int retval = MIXSRC_NONE;
+  for (int i = min; i <= max; i++) {
+    if ((*func)(i)) {
+      retval = i;
+      break;
+      }
```
Comment (bsongis) [inline radio/src/gui/Taranis/helpers.cpp]: Small indentation problem

File: radio/src/gui/Taranis/navigation.cpp [hunk 2]
```diff
@@ -70,6 +71,42 @@ INIT_STOPS(stops100, 3, -100, 0, 100)
...
+{
+  if (result == STR_MENU_INPUTS)
+    s_source = getFirstAvailableSource(MIXSRC_FIRST_INPUT, MIXSRC_LAST_INPUT, isInputAvailable)+1;
+#if defined(LUA_MODEL_SCRIPTS)
+  if (result == STR_MENU_LUA)
+    s_source = getFirstAvailableSource(MIXSRC_FIRST_LUA, MIXSRC_LAST_LUA, isSourceAvailable);
+#endif
+  if (result == STR_MENU_STICKS)
+    s_source = MIXSRC_FIRST_STICK;
+  if (result == STR_MENU_POTS)
```
Comment (bsongis) [inline radio/src/gui/Taranis/navigation.cpp]: else if ...

**Comment by bsongis** [comment]
Otherwise it's all good!

**Comment by mhotar** [comment]
Did someone ask for graphics here?
Full story in five pixels :)

1e05783a-d168-11e4-99d7-277cf591a9e3

1f575f00-d168-11e4-9aba-45d5af644754

**Comment by kilrah** [comment]
> Strange there is no scroll line on the 1st screenshot?

Yes I saw that too... spent some time and I'm _pretty_ sure I invoke the scrollbar correctly, so maybe it's the scrollbar function that has a problem with heights different from the whole body... didn't check that yet

**Comment by bsongis** [comment]
Ok leave it with me! Will you integrate Martin's graphics? Otherwise I will do tonight

File: radio/src/gui/Taranis/navigation.cpp (hunk 2 above)
Comment (kilrah) [inline radio/src/gui/Taranis/navigation.cpp]: Added them... but isn't it just unnecessary clutter?

**Comment by kilrah** [comment]
OK, should be all good now :)

**Comment by bsongis** [comment]
Thanks!

**Comment by kilrah** [comment]
> This was badly needed! The list was getting very loooong... :+1:

Yeah, created an issue for it more than a year ago... but it got particularly obvious it was time to do it the other day when I got an overflow of the uint8_t list :\
https://github.com/opentx/opentx/commit/d54a0389ea31852f259c3643748e987470c440ee

**Pull request merged**